### PR TITLE
EDM-1746: optimize spec comparison

### PR DESCRIPTION
**Pull request by avishayt**
- Replace reflection-based DeviceSpecsAreEqual and FleetSpecsAreEqual with optimized implementation
- Create internal/util/deepequal.go with specialized reflection engine featuring:
  - Type caching to avoid repeated reflection computation
  - Fast paths for common data types (slices, maps, structs)
  - Optimized union type handling for json.RawMessage fields
  - Cycle detection for recursive structures
  - JSON fallback for edge cases ensuring correctness
- Performance improvement: ~19x faster (4.7µs vs 91.6µs per operation)
- Maintain exact semantic compatibility with existing behavior
- Add comprehensive test coverage for edge cases and integration scenarios
- Fix critical FleetSpecsAreEqual bug with map ordering that caused unpredictable failures

## Summary by CodeRabbit

* **New Features**
  * Introduced advanced deep equality checks for device and fleet specifications, improving accuracy in comparing complex configurations.

* **Bug Fixes**
  * Resolved issues where map key ordering and distinctions between nil and empty fields could cause incorrect equality results for device and fleet specs.

* **Tests**
  * Added comprehensive unit and integration tests to validate equality logic, including edge cases and real-world database scenarios.

### CODE REVIEW

File: internal/util/deepequal.go [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+
+// typeInfo caches reflection metadata for types to avoid repeated computation
+type typeInfo struct {
+	hasUnionFields bool  // true if type contains json.RawMessage fields (union types)
+	unionFields    []int // indices of json.RawMessage fields
+}
+
+var (
+	typeCache = make(map[reflect.Type]*typeInfo)
+	cacheMu   sync.RWMutex
```
Comment (hexfusion) [inline internal/util/deepequal.go:17]: is a global mutex a possible issue?
